test/Dialect/SV/hw-export-module-hierarchy.mlir

```diff
@@ -0,0 +1,25 @@
+// RUN: firtool %s --verilog --export-module-hierarchy | FileCheck %s
```

Comment:
Can you pare down this test so that it is only running just this one pass? Something like:

```
circt-opt -pass-pipeline='firrtl.circuit(...) %s | FileCheck %s
```

The test would then need to be updated to look for `sv.verbatim` ops instead of the ExportVerilog-generated strings.

End-to-end integration tests are fantastic and necessary, but it's overkill to run everything that `firtool` is doing just for this. 😅

Reply:
yes, very few tests should be running firtool directly.  Rather, circt-op can run just your pass.

Reply:
I actually tried using the `circt-opt -pass-pipeline` thing earlier, but was having trouble with it. Let me try a bit harder to make it work, but do you know why this command doesn't work?

```sh
$ circt-opt -pass-pipeline='builtin.module(hw-export-module-hierarchy)' test/Dialect/SV/hw-export-module-hierarchy.mlir
module attributes {firrtl.mainModule = "TestHarness"}  {
  hw.module @InnerModule(%in: i1) -> (out: i1) {
    hw.output %in : i1
  }
  hw.module @MainDesign(%in: i1) -> (out: i1) attributes {firrtl.DesignUnderTest} {
    %inner.out = hw.instance "inner" @InnerModule(in: %in: i1) -> (out: i1)
    hw.output %inner.out : i1
  }
  hw.module @TestHarness() {
    %true = hw.constant true
    %main_design.out = hw.instance "main_design" @MainDesign(in: %true: i1) -> (out: i1)
    hw.output
  }
}
```

It basically just prints out the original test file unmodified besides stripping out the magic FileCheck comments.

My top-level module is an `mlir::ModuleOp` because that's where we are currently attaching the `firrtl.mainModule` attribute, but maybe this won't be an issue for me anymore if I switch over to @darthscsi's suggestion of lowering the `firrtl.mainModule` and `firrtl.DesignUnderTest` attributes to a more generic "dump hierarchy to file" attribute.

Reply:
Try `-pass-pipeline='hw-export-module-hierarchy'`.  You don't need to nest pass under the top level mlir module.  If you wanted to run the pass on each `hw.module`, you would do: `-pass-pipeline='hw.module(hw-export-module-hierarchy'`.

Reply:
Oh, duh, that makes perfect sense. Thanks!

Reply:
One other thing which might clarify what is going on here: all `.mlir` files parsed by `circt-opt` have an implicit `mlir::ModuleOp` wrapping everything even if its not spelled out in the file.